### PR TITLE
core(lhr): move runtime config to report, configSettings

### PR DESCRIPTION
**Pull request by patrickhulce**
- removes `runtimeConfig` from LHR
- adds `runtimeSettings` to LHR (I'd prefer just `.settings` but doing the rename of that back to something else would have been more challenging, so lmk if other folks feel the same I'd I'll happily change :D)
- moves the emulation description logic to the report renderer

I know that this whole thing is ripe for some refactoring, but we've got a long list to get through and this keeps same functional parity, sooooo

image

NOTE: closure fixes pending

### CODE REVIEW

File: lighthouse-core/test/results/sample_v2.json [hunk 1]
```diff
@@ -4546,23 +4546,28 @@
...
-        "description": "562.5 ms HTTP RTT, 1474.5600000000002 Kbps down, 675 Kbps up (DevTools)"
-      },
-      {
-        "name": "CPU Throttling",
-        "description": "4x slowdown (DevTools)"
-      }
-    ],
-    "blockedUrlPatterns": [],
-    "extraHeaders": {}
+  "runtimeSettings": {
```
Comment (paulirish) [inline lighthouse-core/test/results/sample_v2.json]: wdyt `configSettings`? that's what we call them elsewhere.

Comment (patrickhulce) [reply lighthouse-core/test/results/sample_v2.json]: sg done

File: lighthouse-core/test/results/sample_v2.json [hunk 2]
```diff
@@ -4432,23 +4432,27 @@
...
-        "description": "562.5 ms HTTP RTT, 1474.5600000000002 Kbps down, 675 Kbps up (DevTools)"
-      },
-      {
-        "name": "CPU Throttling",
-        "description": "4x slowdown (DevTools)"
-      }
-    ],
-    "blockedUrlPatterns": [],
-    "extraHeaders": {}
+  "configSettings": {
```
Comment (brendankenny) [inline lighthouse-core/test/results/sample_v2.json]: this is going to be great to have

**Comment by brendankenny** [comment]
> I know that this whole thing is ripe for some refactoring

did you mean in how we generate things in `getEmulationDesc` or more? Make sure to make a note of it :)

**Pull request merged**